tests/test_config_command.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use regex::Regex;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+#[test]
+fn test_config_list_single() {
+    // TODO: Populate expected config values explicitly in this file.
+    let test_env = TestEnvironment::default();
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "user.name"]);
+    insta::assert_snapshot!(stdout, @r###"
+    user.name="Test User"
+    "###);
+}
+
+#[test]
+fn test_config_list_table() {
+    let test_env = TestEnvironment::default();
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "user"]);
+    insta::assert_snapshot!(
+        find_stdout_line("user.name", &stdout),
+        @r###"user.name="Test User""###);
+    insta::assert_snapshot!(
+        find_stdout_line("user.email", &stdout),
+        @r###"user.email="test.user@example.com""###);
```

Comment:
If you use `test_env.add_config()` as mentioned above, then you should be able to `assert_snapshot()` the full output. That would be useful for checking that the order is predictable and reasonable (probably alphabetical?).

Reply:
In my testing the order wasn't deterministic. 🙁 Was hoping to improve that next.

The ideal would be in definition/precedence order, not purely alphabetical.

Reply:
> In my testing the order wasn't deterministic. 🙁 Was hoping to improve that next.

Sounds good.

> The ideal would be in definition/precedence order, not purely alphabetical.

Good point. I don't think the `config` crate currently gives us access to that information, but I suspect it will be available after their rewrite. I commented about that on https://github.com/mehcode/config-rs/issues/324#issuecomment-1365352055, so they're aware of the usecase.

Reply:
Thanks! For now I did a little manual sorting in the tests (just before asserting on output) w/ TODOs to get rid of the reordering when possible.